src/main/java/hello/HelloController.java

```diff
@@ -3,11 +3,26 @@
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.bind.annotation.RequestMapping;
 
+import javax.servlet.http.HttpServletRequest;
+
 @RestController
-public class HelloController {
-    
+public class HelloController implements Comparable {
+    public static String FOO_VAR = "";
+
     @RequestMapping("/")
-    public String index() {
+    public String index(HttpServletRequest request) {
         return "Greetings from Spring Boot!";
     }
-}
+
+    public Object clone() {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this "clone" implementation; use a copy constructor or copy factory instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.35.192.52.128.xip.io/coding_rules#rule_key=squid%3AS2975)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use super.clone() to create and seed the cloned instance to be returned. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.35.192.52.128.xip.io/coding_rules#rule_key=squid%3AS1182)